iota-client/src/builder.rs

```diff
@@ -128,21 +128,27 @@ impl ClientBuilder {
         let sync = Arc::new(RwLock::new(HashSet::new()));
         let sync_ = sync.clone();
 
-        let stop_sync = Arc::new(AtomicBool::new(false));
-        let stop_sync_ = stop_sync.clone();
+        let (sync_kill_sender, sync_kill_receiver) = channel(1);
```

Comment:
I am not sure it really needs to be a `broadcast`, `oneshot` if made for this purpose and probably sufficient ?

Reply:
I tried with an oneshot channel but I think I didn't understand its usage with tokio::select. First of all the try_recv method isn't async so the behavior is different. Then it also panics saying that I read it after it was complete.